### PR TITLE
Adding server span generation on receiving subscribe and binding event

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Adding server span generation on receiving subscribe and binding event

Done as part of https://github.com/dapr/dapr/pull/1545

## Issue reference
Part of Epic 
#292 
#1373 
#1533
## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -939,13 +956,19 @@ func (a *DaprRuntime) publishMessageHTTP(msg *pubsub.NewMessage) error {
 	req.WithHTTPExtension(nethttp.MethodPost, "")
 	req.WithRawData(msg.Data, pubsub.ContentType)
 
-	// TODO Propagate Context
 	ctx := context.Background()
+	spanName := fmt.Sprintf("SubscribedMessage: %s", msg.Topic)
```
Comment (yaron2) [inline pkg/runtime/runtime.go:960]: This is a wrong span name. This is when publishing a message to the app, I think `SubscribeMessage` is either wrong or confusing.

Same for the other instance below.

**Comment by yaron2** [comment]
Also, in general, I keep seeing the span creation being duplicated and repeated everywhere in our code, while we know this needs to change.

File: pkg/runtime/runtime.go (hunk 1 above)
Comment (shalabhms) [reply pkg/runtime/runtime.go]: Changed it to PublishMessage. Let me know if this is fine.

**Comment by shalabhms** [comment]
> Also, in general, I keep seeing the span creation being duplicated and repeated everywhere in our code, while we know this needs to change.

We need to create span or set span ( client/server ) , in the first incoming call to Dapr and outgoing call to app.

**Comment by yaron2** [comment]
> > Also, in general, I keep seeing the span creation being duplicated and repeated everywhere in our code, while we know this needs to change.
> 
> We need to create span or set span ( client/server ) , in the first incoming call to Dapr and outgoing call to app. We can think of creating a common layer before the routing handler - probably in next milestone.

We have to do something about it in the next milestone, we currently have too many create/set spans, also in non-entrypoint or exit methods. And those should not occur if the sampling rate is 0.

/cc @youngbupark

**Comment by shalabhms** [comment]
> f the sampling rate is 0.

Yes, we will disable spans once sampling rate is 0. Tracking #1473

**Comment by shalabhms** [comment]
Moving this PR to 0.8.0 milestone.  We can merge the PR once we have decision on publish/subscribe and bindings trace flow.

cc @yaron2 @youngbupark

**Comment by yaron2** [comment]
Thanks, I think this is the right call

**Comment by youngbupark** [comment]
@yaron2 I am sure that the total number of span generation frequency is lower than the previous code because the previous tracing span were generated for every api call. We will talk about it more in the next meeting.

**Comment by shalabhms** [comment]
Merged as part of #1533 and PR https://github.com/dapr/dapr/pull/1545

**Pull request closed without merging**